drivers/entropy/entropy_ll_nrfxlib.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <entropy.h>
+#include <soc.h>
+#include <device.h>
+#include <irq.h>
+#include <kernel.h>
+#include <kernel_includes.h>
+#include <ble_controller.h>
+#include <ble_controller_soc.h>
+
+#include <multithreading_lock.h>
+
+#define DEV_DATA(dev) \
+	((struct rng_driver_data *)(dev)->driver_data)
+
+/* The RNG driver data. */
+struct rng_driver_data {
+	/* Used to fill the pending client buffer with new random values after
+	 * RNG IRQ is triggered.
+	 */
+	struct k_sem sem_sync;
+};
+
+static struct rng_driver_data rng_data;
+
+static int rng_driver_get_entropy(struct device *dev, u8_t *buf, u16_t len)
+{
+	__ASSERT_NO_MSG(&rng_data == DEV_DATA(dev));
+
+	u8_t *p_dst = buf;
+	u16_t bytes_left = len;
+
+	while (bytes_left > 0) {
+		s32_t bytes_read = 0;
+
+		while (bytes_read <= 0) {
+			s32_t errcode = MULTITHREADING_LOCK_ACQUIRE();
+
+			if (errcode) {
+				return errcode;
+			}
+			bytes_read = ble_controller_rand_vector_get(p_dst,
```

Comment:
where is `ble_controller_rand_vector_get` defined? mocking code below suggests it returns `u32_t` but you have bytes read as `s32_t` which would suggest function returning an error with negative value. There is a check in loop to see if `bytes_read <= 0`... so can it be negative? at the same time there is no error checking as the only check is `if (!bytes_read) {`...

Reply:
https://github.com/NordicPlayground/nrfxlib/blob/master/ble_controller/include/ble_controller_soc.h#L122
`bytes_read` should be changed to `u32_t`.